src/lib.rs

```diff
@@ -1,4 +1,13 @@
 #![doc = include_str!("../README.md")]
+#![deny(
+    unknown_lints,
+
+    clippy::all,
+    clippy::perf,
+
+    clippy::disallowed_script_idents,
+    clippy::fn_to_numeric_cast_any,
+)]
```

Comment:
Whoops! I forgot to add what these do. Basically, it now shows default clippy lints as errors, and performance lints show as errors as well, because *sped*. The last two Clippy rules are basically "don't use anything besides Latin and Cyrillic for names" and "don't cast function pointers as numbers", both felt kind of important so I put them there.

Reply:
As mentioned in my main comment - none of these lints should be deny as it causes breaking API changes for downstream users when lints change, which given that I'm aiming for a stable 1.0.0 release someday that follows semver, I'd like to avoid.

It's also not clear what benefit "all" lints would provide. I've already added the most common best practice and idoms lints and anything beyond that is getting fairly pedantic and opinionated though perf may be a good choice for this particular crate - though I haven't had a chance to review what lints it provides.